designer/server/src/main/scala/pl/touk/nussknacker/ui/api/ManagementResources.scala

```diff
@@ -243,18 +244,20 @@ class ManagementResources(val processAuthorizer: AuthorizeProcess,
               val params = req.params.getOrElse(Map.empty)
               complete {
                 customActionInvokerService.invokeCustomAction(req.actionName, process, params)
-                  .flatMap {
-                    case res@Right(_) =>
+                  .transformWith {
```

Comment:
I dont like this part. Why dont' you use  flatMap + recoverWith ?  It  allows one to separate business logic from error handing and I believe you dont have to care about this generic case `case Failure(err) =>` at all.

Reply:
You can do it even better / easier. Specific errors should implement specific interfaces and error code mapping should be done by EspErrorToHttp.espErrorToHttp -> I'm not sure, but I think instead of returning Future.failed, you can just throw an exception as well.

Reply:
@deikare the point of this change was to remove this piece of code and use `EspErrorToHttp` instead :)